### ISSUE TITLE
lxqt-config build failed with libxcb 1.12

### ISSUE THREAD

**Issue by OhDaeto** [lxde/lxqt#1059]
GCC version and options: https://github.com/lxde/qtermwidget/issues/65
```
[ 91%] Built target lxqt-config-brightness_automoc
[ 91%] Generating lxqt-config-brightness.desktop
[ 92%] Generating lxqt-config-brightness_ru.qm
Updating '/tmp/yaourt-tmp-narwhal/aur-lxqt-config-git/src/build/lxqt-config-brightness/lxqt-config-brightness_ru.qm'...
    Generated 3 translation(s) (3 finished and 0 unfinished)
    Ignored 1 untranslated source text(s)
Scanning dependencies of target lxqt-config-brightness
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/main.cpp.o
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/brightnesssettings.cpp.o
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o
/tmp/yaourt-tmp-narwhal/aur-lxqt-config-git/src/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp: In member function 'void XRandrBrightness::gamma_brightness_set(xcb_randr_output_t, float)':
/tmp/yaourt-tmp-narwhal/aur-lxqt-config-git/src/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp:246:23: warning: variable 'void_cookie' set but not used [-Wunused-but-set-variable]
     xcb_void_cookie_t void_cookie = xcb_randr_set_crtc_gamma (QX11Info::connection(), output_info->crtc, red_length, red, green, blue);
                       ^~~~~~~~~~~
/tmp/yaourt-tmp-narwhal/aur-lxqt-config-git/src/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp: In member function 'void XRandrBrightness::setMonitorsSettings(QList<MonitorInfo>)':
/tmp/yaourt-tmp-narwhal/aur-lxqt-config-git/src/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp:384:29: warning: variable 'backlight_max' set but not used [-Wunused-but-set-variable]
         long cur, min, max, backlight_max = -1, backlight_value = 0;
                             ^~~~~~~~~~~~~
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/monitorinfo.cpp.o
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/outputwidget.cpp.o
[ 95%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/lxqt-config-brightness_automoc.cpp.o
[ 95%] Linking CXX executable lxqt-config-brightness
/usr/bin/ld: CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o: undefined reference to symbol 'xcb_get_setup'
/usr/lib/libxcb.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/build.make:252: lxqt-config-brightness/lxqt-config-brightness] Error 1
make[1]: *** [CMakeFiles/Makefile2:708: lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/all] Error 2
make: *** [Makefile:150: all] Error 2

**Comment by palinek** [comment]
@OhDaeto can you check if the https://github.com/lxde/lxqt-config/commit/93672a4022ec9d03b4f2de7c6b653da747c4c4fd in [branch brightness_flags](https://github.com/lxde/lxqt-config/tree/brightness_flags) makes any change?

**Comment by OhDaeto** [comment]
No. 
```
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/monitorinfo.cpp.o
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/outputwidget.cpp.o
[ 95%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/lxqt-config-brightness_automoc.cpp.o
[ 95%] Linking CXX executable lxqt-config-brightness
/usr/bin/ld: CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o: undefined reference to symbol 'xcb_get_setup'
/usr/lib/libxcb.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/build.make:252: lxqt-config-brightness/lxqt-config-brightness] Error 1
make[1]: *** [CMakeFiles/Makefile2:708: lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/all] Error 2
make: *** [Makefile:150: all] Error 2

**Comment by palinek** [comment]
Is the xcb linked against (`-lxcb` in link command)? -> please, test with `make VERBOSE=1`

If not, is the `xcb` required by `xcb-randr`? -> please, paste contents of /usr/lib/pkgconfig/xcb-randr.pc

**Comment by pmattern** [comment]
Output running `make VERBOSE=1`, same context as depicted above, all findings made by @OhDaeto so far confirmed
```
[ 95%] Linking CXX executable lxqt-config-brightness
cd /usr/local/peter/Tests/lxqt-config/Projekt/build/lxqt-config-brightness && /usr/bin/cmake -E cmake_link_script CMakeFiles/lxqt-config-brightness.dir/link.txt --verbose=1
/usr/bin/c++    -fno-exceptions -Wall -std=c++11   CMakeFiles/lxqt-config-brightness.dir/main.cpp.o CMakeFiles/lxqt-config-brightness.dir/brightnesssettings.cpp.o CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o CMakeFiles/lxqt-config-brightness.dir/monitorinfo.cpp.o CMakeFiles/lxqt-config-brightness.dir/outputwidget.cpp.o CMakeFiles/lxqt-config-brightness.dir/lxqt-config-brightness_automoc.cpp.o  -o lxqt-config-brightness -rdynamic -lxcb-randr /usr/lib/liblxqt.so.0.10.0 /usr/lib/libKF5WindowSystem.so.5.22.0 /usr/lib/libQt5X11Extras.so.5.6.0 /usr/lib/libQt5Xdg.so.1.3.0 /usr/lib/libQt5Widgets.so.5.6.0 /usr/lib/libQt5DBus.so.5.6.0 /usr/lib/libQt5Xml.so.5.6.0 /usr/lib/libQt5XdgIconLoader.so.1.3.0 /usr/lib/libQt5Gui.so.5.6.0 /usr/lib/libQt5Core.so.5.6.0
/usr/bin/ld: CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o: undefined reference to symbol 'xcb_get_setup'
/usr/lib/libxcb.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/build.make:251: recipe for target 'lxqt-config-brightness/lxqt-config-brightness' failed
make[2]: *** [lxqt-config-brightness/lxqt-config-brightness] Error 1
make[2]: Leaving directory '/usr/local/peter/Tests/lxqt-config/Projekt/build'
CMakeFiles/Makefile2:707: recipe for target 'lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/all' failed
make[1]: *** [lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/all] Error 2
make[1]: Leaving directory '/usr/local/peter/Tests/lxqt-config/Projekt/build'
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```
So a switch `-lxcb-randr` is set here while the equivalent invocations of `lxqt-config-cursor`, `lxqt-config-input` and `lxqt-config-appearance` are stating `-lxcb` only.
Content of `xcb-randr.pc`
```
# cat /usr/lib/pkgconfig/xcb-randr.pc 
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: XCB RandR
Description: XCB RandR Extension
Version: 1.12
Requires.private: xcb xcb-render
Libs: -L${libdir} -lxcb-randr
Cflags: -I${includedir}
```

**Comment by palinek** [comment]
Updated the  [brightness_flags branch](https://github.com/lxde/lxqt-config/tree/brightness_flags). Could you, @pmattern, @OhDaeto, check that?

**Comment by pmattern** [comment]
Using lxde/lxqt-config@d34c891 the problem seems to boil down to some warning messages
```
[...]
Scanning dependencies of target lxqt-config-brightness
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/main.cpp.o
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/brightnesssettings.cpp.o
[ 93%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/xrandrbrightness.cpp.o
/usr/local/peter/Tests/lxqt-config/Projekt/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp: In member function ‘void XRandrBrightness::gamma_brightness_set(xcb_randr_output_t, float)’:
/usr/local/peter/Tests/lxqt-config/Projekt/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp:246:23: warning: variable ‘void_cookie’ set but not used [-Wunused-but-set-variable]
     xcb_void_cookie_t void_cookie = xcb_randr_set_crtc_gamma (QX11Info::connection(), output_info->crtc, red_length, red, green, blue);
                       ^~~~~~~~~~~
/usr/local/peter/Tests/lxqt-config/Projekt/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp: In member function ‘void XRandrBrightness::setMonitorsSettings(QList<MonitorInfo>)’:
/usr/local/peter/Tests/lxqt-config/Projekt/lxqt-config/lxqt-config-brightness/xrandrbrightness.cpp:384:29: warning: variable ‘backlight_max’ set but not used [-Wunused-but-set-variable]
         long cur, min, max, backlight_max = -1, backlight_value = 0;
                             ^~~~~~~~~~~~~
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/monitorinfo.cpp.o
[ 94%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/outputwidget.cpp.o
[ 95%] Building CXX object lxqt-config-brightness/CMakeFiles/lxqt-config-brightness.dir/lxqt-config-brightness_automoc.cpp.o
[ 95%] Linking CXX executable lxqt-config-brightness
[ 96%] Built target lxqt-config-brightness
Scanning dependencies of target lxqt-config-locale_automoc
[...]
```
and `make` completes successfully.

**Comment by palinek** [comment]
fixed by lxde/lxqt-config@d268e8d4964a4676290a1d2a87826e5aa747ca72
